### DISCUSSION TITLE
AusLock - Sciener - TTLock

### DISCUSSION THREAD

**Discussion by 49Penrose** [jbergler/hass-ttlock#132] [Q&A] [answered]
Hi there HA newbie, in Australia and installed AusLock Smart Locks, have working via Alexa (ScienerSmart Skill). I have created an account on the TTLock Open Platform , Do I need to make an app for this integration ? I dont understand cant I just use your app? Ive download your HA Integration and at the stage of entering credentials , Name , OAuthClient ID , OAuthCLient Secret. How to I get these credentials?

**Comment by jbergler** [comment]
Hiya - thanks for using the discussions rather than filing an issue.

Unfortunately you do need to create your own app. The reason why is that webhooks can only be configured per app, which means if there was a shared one I'd have to build & run some infrastructure in the cloud with access to everyones locks. That's both expensive in terms of time and money, plus it's a big responsibility to secure properly.

Creating an app is pretty straightforward, and should only take a few minutes (plus a day or so to wait for your app to be approved, if they still require that step).

Once you have the app, it will give you the oauth credentials, and you should be able to continue setting it up

**Comment by 49Penrose** [comment]
Thanks for the reply Jonas,I have no idea how to create an app, is there a template I can use and just change for my webhooks?Thanks in advance

**Reply by jbergler** [reply]
FYI, I edited this because your email signature had your phone number in it. Not sure you wanted that shared.

**Comment by jbergler** [comment, marked as answer]
Did you see the readme for the integration? It has the steps documented.
https://github.com/jbergler/hass-ttlock?tab=readme-ov-file#creating-an-oauth-app

**Reply by 49Penrose** [reply]
yes i have followed the steps, I am at this stage of TTLockOpen Platform (see attached) do I just fill in the fields with any name, discription and image?
Then wait ? 
I'm at the stage pictured on the Integration on HA
![LockApp](https://github.com/user-attachments/assets/82022e63-174f-4300-9c48-fe026890d87b)
![githubsnip](https://github.com/user-attachments/assets/ab974ffd-78a3-4962-aec4-81cde235c8ab)

**Reply by jbergler** [reply]
Yep correct. You can call it "home assistant" or whatever else you like, the values don't matter as long as you leave "app" checked.

**Reply by 49Penrose** [reply]
Thank you, I have submitted App and now "in review" appreciate your help. Hopefully once the App credentials come through I can get the AusLocks working and will let you know 
Thanks Again

**Reply by 49Penrose** [reply]
Hi Jonas thanks for the help, just wanted to let you know the the integration is working for AusLocks Smart Locks. I havent fully tested but has integrated Lock Name and Battery info as shown in the native app. link for auslocks
https://www.auslock.net/

**Reply by jbergler** [reply]
Great news, thanks!
